src/librustc_mir/borrow_check/mod.rs

```diff
@@ -897,6 +976,55 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
                     .enumerate()
                     .for_each(|field| drop_field(self, field));
             }
+
+            // #45696: special-case Box<T> by treating its dtor as
+            // only deep *across owned content*. Namely, we know
+            // dropping a box does not touch data behind any
+            // references it holds (which a Deep Write would touch).
+            ty::TyAdt(def, _) if def.is_box() => {
+                // First, we model the loss of the reference itself
+                // via a shallow write.
```

Comment:
"the loss of the reference itself"?

Reply:
That is, the dropping of the representation of the box-pointer

How would you prefer to phrase this?

Reply:
Do you mean the allocation? Or the library-side newtype around the pointer? I would not call either a "reference" unless you mean the `&mut Box<T>` that a `Drop::drop` impl would get.

Reply:
I mean the library-side newtype around the pointer.  I tried rephrasing as "StorageDead of the box-pointer representation" in my updated comments.